Move document from g/documentation (#379)
gardener/gardener-extension-provider-azure: docs/tutorials/kubernetes-cluster-in-azure-with-gardener/_index.md

---
title: Create shoot clusters in Azure
level: beginner
category: Getting Started
scope: app-developer
---

# Create shoot clusters in Azure

### Prerequisites

> -   You need an Azure account.

Reviewer: You need to have as well as Azure Service Principal assigned to your subscription or the permissions to create one in the Active Directory tenant and assign it to your subscription.

-   You have access to the Gardener dashboard and have permissions to create projects.

1. Go to the Gardener dashboard and create a *Project*.

    <img src="images/new_gardener_project.jpg">

> 2. Get properties of your Azure account/Service Principal.

Reviewer: ```suggestion
2. Get properties of your Azure AD tenant, Subscription and Service Principal.
```

    ```
    Before you can provision and access a Kubernetes cluster on Azure, you need to add the account credentials in Gardener. 
    Gardener needs the credentials to provision and operate the Azure infrastructure for your Kubernetes cluster.

    **Ensure that the account has the `contributor` role.**
    ```

    -  Tenant ID
    

>         The TenantID is also called DirectoryID - https://portal.azure.com/#blade/Microsoft_AAD_IAM/ActiveDirectoryMenuBlade/Properties.

Reviewer: I guess the term `DirectoryID` is not a valid one. 
Can we remove the portal link here?

Could you refer here to the official Azure docs: https://docs.microsoft.com/en-us/azure/active-directory/fundamentals/active-directory-how-to-find-tenant

        <img src="images/azuregettenantid.jpg">

    -  ClientID
        Select the subscription.
        <img src="images/azureselectsubscription.jpg">

    - Select the SPN.
        <img src="images/azureselectspn.jpg">

>     *Note:* The ClientID is also called ApplicationID.

Reviewer: ```suggestion
    *Note:* A service principal consist of a `ClientID` (also called `ApplicationID`) and a Client Secret
```

We could also refer here to the official Azure docs: https://docs.microsoft.com/en-us/azure/active-directory/develop/app-objects-and-service-principals

    <img src="images/azuregetclientid.jpg">

    - Client Secret
        Secrets for the Azure Account/Service Principal can be genereted/rotated via the Azure Portal.
        Access the [Azure Portal](https://portal.azure.com) and navigate to the `Active Directory` service.
        Within the service navigate to `App registrations` and select your service principal. 
        In the detail view navigate to `Certificates & secrets`. In the section, you can generate a new secret for the Service Principal.

> 3. On the Gardener dashboard, choose *Secrets* and then the plus sign <img src="images/plus_icon.jpg"> in the Azure frame to add a new Azure secret.

Reviewer: I guess the `plus_icon.jpg` is not really required.

    <img src="images/gardenernewazuresecret.jpg">

> 4. Provide the details for the Azure service account.  
>     After processing the ticket, you’ll receive the Service Principle credentials via email.
>     Copy "Key Value" from the email into "Client Secret".

Reviewer: ```suggestion
4. Enter the `TenantID`, `SubscriptionID` and the Service Principal credentials (`ClientID` and `ClientSecret`) into the Secret definition.
```

    <img src="images/gardeneraddazuresecret.jpg">

5. To create a new cluster, choose *Clusters* and then the plus sign in the lower right corner.

>     <img src="images/new_cluster.jpg">

Reviewer: I guess this picture is outdated. Can we provide new one?

> ---

Reviewer 2: I don't see `images/gardenerazuresecret.jpg` being linked/used by this tutorial. Can we remove it or make the tutorial use it?

> ---

Reviewer 2: I don't see `images/gardenerazuresecret0.jpg` being linked/used by this tutorial. Can we remove it or make the tutorial use it?

title: Create shoot clusters in Azure
level: beginner
category: Getting Started
scope: app-developer
---

# Create shoot clusters in Azure

### Prerequisites

-   You need an Azure account.
-   You have access to the Gardener dashboard and have permissions to create projects.

Before you can provision and access a Kubernetes cluster on Azure, you need to add the account credentials in Gardener. 
Gardener needs the credentials to provision and operate the Azure infrastructure for your Kubernetes cluster.

> Ensure that the account has the **contributor** role.

1. Request a new service account on Azure. 

    You must [request a new service account](https://jira.multicloud.int.sap/plugins/servlet/desk/portal/2) first, if you don't have one.

    <img src="images/request_spn.jpg">

    Naming Convention for the Service UserID: `azrspn_<PartOfSubName>_usecase`

>     <img src="images/request_spn1.jpg">

Reviewer 2: I think this step should be reworked as it describes SAP specific process.

    -  Tenant ID
    
        The TenantID is also called DirectoryID - https://portal.azure.com/#blade/Microsoft_AAD_IAM/ActiveDirectoryMenuBlade/Properties.
        <img src="images/azuregettenantid.jpg">

    -  ClientID
        Select the subscription.
        <img src="images/azureselectsubscription.jpg">

    - Select the SPN.
        <img src="images/azureselectspn.jpg">

    *Note:* The ClientID is also called ApplicationID.
    <img src="images/azuregetclientid.jpg">

    - Client Secret
        Secrets for the Azure Account/Service Principal can be genereted/rotated via the Azure Portal.
        Access the [Azure Portal](https://portal.azure.com) and navigate to the `Active Directory` service.
        Within the service navigate to `App registrations` and select your service principal. 
        In the detail view navigate to `Certificates & secrets`. In the section, you can generate a new secret for the Service Principal.

3. On the Gardener dashboard, choose *Secrets* and then the plus sign <img src="images/plus_icon.jpg"> in the Azure frame to add a new Azure secret.

    <img src="images/gardenernewazuresecret.jpg">

> 5. Provide the details for the Azure service account.  

Reviewer 2: Right now there is not step 4, the tutorial is jumping from step 3. to 5.

```suggestion
4. Provide the details for the Azure service account. 
```
